webui/src/data/server.js

```diff
@@ -285,11 +291,30 @@ class ServerCache {
 
             languages: null,
             disciplines: null,
+
+            currentCommunityRoles: {}, // a dictionary of current community's roles name and ID
```

Comment:
It's better not to count on anything being 'current', so let's make this a dictionary of communities to dictionary of roles to users, e.g.: 

communityRoles: {
    role: {user, user, user}
}